From c9522b213d62b470ca596a7859b2413f1e3eebf9 Mon Sep 17 00:00:00 2001
From: Alexander Larsson <alexl@redhat.com>
Date: Fri, 11 Dec 2015 14:37:56 +0100
Subject: [PATCH] Remove python3 optimized mode by default

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 02bce12..7ed8a80 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@ ID=org.freedesktop
 VERSION=1.2
 ARCH=x86_64
 IMAGEDIR=freedesktop-sdk-base/images/$(ARCH)
-BASE_HASH=52853d3961aeba5e24cfb251422c2d0842340039
+BASE_HASH=4f89712a14cf936d2afa6e8abae4a02738196c14
 
 EXTRA_NAME=
 DELTAS=
-- 
2.5.0

