diff -up cfe-3.7.0.src/lib/Driver/Tools.cpp.dynamic_linker cfe-3.7.0.src/lib/Driver/Tools.cpp
--- cfe-3.7.0.src/lib/Driver/Tools.cpp.dynamic_linker	2015-09-22 14:58:46.078350528 +0200
+++ cfe-3.7.0.src/lib/Driver/Tools.cpp	2015-09-22 14:59:03.204096047 +0200
@@ -8015,7 +8015,7 @@ static std::string getLinuxDynamicLinker
            ToolChain.getTriple().getEnvironment() == llvm::Triple::GNUX32)
     return "/libx32/ld-linux-x32.so.2";
   else
-    return "/lib64/ld-linux-x86-64.so.2";
+    return "/lib/ld-linux-x86-64.so.2";
 }
 
 static void AddRunTimeLibs(const ToolChain &TC, const Driver &D,
diff -up cfe-3.7.0.src/test/Driver/dyld-prefix.c.dynamic_linker cfe-3.7.0.src/test/Driver/dyld-prefix.c
--- cfe-3.7.0.src/test/Driver/dyld-prefix.c.dynamic_linker	2015-09-22 15:02:47.437764042 +0200
+++ cfe-3.7.0.src/test/Driver/dyld-prefix.c	2015-09-22 15:02:50.198723016 +0200
@@ -6,7 +6,7 @@
 // CHECK-32: "-dynamic-linker" "/foo/lib/ld-linux.so.2"
 
 // RUN: %clang -target x86_64-unknown-linux --dyld-prefix /foo -### %t.o 2>&1 | FileCheck --check-prefix=CHECK-64 %s
-// CHECK-64: "-dynamic-linker" "/foo/lib64/ld-linux-x86-64.so.2"
+// CHECK-64: "-dynamic-linker" "/foo/lib/ld-linux-x86-64.so.2"
 
 // RUN: %clang -target x86_64-unknown-linux-gnux32 --dyld-prefix /foo -### %t.o 2>&1 | FileCheck --check-prefix=CHECK-X32 %s
 // CHECK-X32: "-dynamic-linker" "/foo/libx32/ld-linux-x32.so.2"
